apps/src/templates/teacherDashboard/AddSectionDialog.jsx

```diff
@@ -12,86 +12,109 @@ import {
   beginImportRosterFlow,
   setRosterProvider,
   editSectionProperties,
-  cancelEditingSection
+  cancelEditingSection,
+  assignedCourseOffering
 } from './teacherSectionsRedux';
 import ParticipantTypePicker from './ParticipantTypePicker';
+import Spinner from '@cdo/apps/code-studio/pd/components/spinner';
 
 /**
  * UI for a teacher to add a new class section.  For editing a section see
  * EditSectionDialog.
  */
-class AddSectionDialog extends Component {
-  static propTypes = {
-    // Provided by Redux
-    isOpen: PropTypes.bool.isRequired,
-    section: sectionShape,
-    beginImportRosterFlow: PropTypes.func.isRequired,
-    setRosterProvider: PropTypes.func.isRequired,
-    setLoginType: PropTypes.func.isRequired,
-    setParticipantType: PropTypes.func.isRequired,
-    handleCancel: PropTypes.func.isRequired,
-    availableParticipantTypes: PropTypes.arrayOf(PropTypes.string).isRequired
+const AddSectionDialog = ({
+  isOpen,
+  section,
+  beginImportRosterFlow,
+  setRosterProvider,
+  setLoginType,
+  setParticipantType,
+  handleCancel,
+  availableParticipantTypes,
+  assignedCourseOffering,
+  asyncLoadComplete
+}) => {
+  useEffect(() => {
+    if (
+      assignedCourseOffering &&
+      asyncLoadComplete &&
+      !section?.participantType
+    ) {
+      setParticipantType(assignedCourseOffering.participant_audience);
+    }
+  }, [assignedCourseOffering, asyncLoadComplete, section?.participantType]);
+
+  const {loginType, participantType} = section || {};
+  const title = i18n.newSectionUpdated();
+
+  const getDialogContent = () => {
+    if (!asyncLoadComplete) {
+      return <Spinner size="large" style={{padding: 50}} />;
+    }
+    /*
+    The Participant Type Picker will be skips if someone only have permissions to create sections for one
+    type of participants. See teacherSectionsRedux for more details on how this is set up.
+    */
+    if (!participantType) {
```

Comment:
I wish there were a more easily understandable phrase than participantType... line 41 is really hard to read and intuitively know what is going on

Reply:
`participantType` is a key in `section`. The series of negated if statements is kind of confusing to me too but I'm not sure how to make it better. We do want it to essentially be:
- if we haven't loaded data, show the spinner
- if we have loaded data but don't know the participant type, show the `ParticipantTypePicker`
- if have loaded data and do know the participant type, but don't know the login type, show the `LoginTypePicker`
- else show the `EditSectionForm`

I'm definitely open to rearranging this if we can think of a way